ksqldb-api-client/src/test/java/io/confluent/ksql/api/client/integration/ClientIntegrationTest.java

```diff
@@ -657,22 +679,111 @@ public void shouldInsertInto() throws Exception {
   }
 
   @Test
-  public void shouldHandleErrorResponseFromInsertInto() {
+  public void shouldPullQueryTableAfterInsert() throws Exception {
     // Given
     final KsqlObject insertRow = new KsqlObject()
-        .put("K", new KsqlObject().put("F1", new KsqlArray().add("my_key")))
+        .put("K", "my_key")
         .put("LONG", 11L);
 
     // When
+    client.insertInto(TEST_TABLE, insertRow).get();
+
+    // Then: cannot pull query table
+    final String query = "SELECT * from " + TEST_TABLE + " WHERE K='my_key';";
     final Exception e = assertThrows(
         ExecutionException.class, // thrown from .get() when the future completes exceptionally
-        () -> client.insertInto(AGG_TABLE, insertRow).get()
+        () -> client.executeQuery(query).get()
     );
 
     // Then
     assertThat(e.getCause(), instanceOf(KsqlClientException.class));
     assertThat(e.getCause().getMessage(), containsString("Received 400 response from server"));
-    assertThat(e.getCause().getMessage(), containsString("Cannot insert into a table"));
+    assertThat(e.getCause().getMessage(), containsString("table isn't queryable"));
+  }
+
+  @Test
+  public void shouldPushQueryTableAfterInsert() throws Exception {
+    // Given
+    final Map<String, Object> properties = new HashMap<>();
+    properties.put("auto.offset.reset", "earliest");
+
+    final KsqlObject insertRow = new KsqlObject()
+        .put("K", "my_key")
+        .put("LONG", 11L);
+
+    // When
+    final String query = "SELECT * from " + TEST_TABLE + " WHERE K='my_key' EMIT CHANGES LIMIT 1;";
+    StreamedQueryResult queryResult = client.streamQuery(query, properties).get();
+
+    final Row row = assertThatEventually(() -> {
```

Comment:
Does `assertThatEventually` apply any timeout?

Reply:
Yes. It has default timeout:

https://github.com/confluentinc/ksql/blob/dd36ca7728103f9fad6fd83d797f70b52ffc5bd0/ksqldb-test-util/src/main/java/io/confluent/ksql/test/util/AssertEventually.java#L101